src/components/Lightbox/index.tsx

```diff
@@ -31,35 +33,59 @@ type LightboxProps = {
     /** Additional styles to add to the component */
     style?: StyleProp<ViewStyle>;
 
-    /** The index of the carousel item */
-    index?: number;
-
-    /** The index of the currently active carousel item */
-    activeIndex?: number;
-
-    /** Whether the Lightbox is used within a carousel component and there are other sibling elements */
-    hasSiblingCarouselItems?: boolean;
-
     /** Range of zoom that can be applied to the content by pinching or double tapping. */
     zoomRange?: Partial<ZoomRange>;
 };
 
 /**
  * On the native layer, we use a image library to handle zoom functionality
  */
-function Lightbox({
-    isAuthTokenRequired = false,
-    uri,
-    onScaleChanged,
-    onError,
-    style,
-    index = 0,
-    activeIndex = 0,
-    hasSiblingCarouselItems = false,
-    zoomRange = DEFAULT_ZOOM_RANGE,
-}: LightboxProps) {
+function Lightbox({isAuthTokenRequired = false, uri, onScaleChanged: onScaleChangedProp, onError, style, zoomRange = DEFAULT_ZOOM_RANGE}: LightboxProps) {
     const StyleUtils = useStyleUtils();
 
+    /** React hooks must be used in the render function of the component at top-level and unconditionally.
+     * Therefore, in order to provide a default value for "isPagerScrolling" if the "AttachmentCarouselPagerContext" is not available,
+     * we need to create a shared value that can be used in the render function.
+     * */
+    const isPagerScrollingFallback = useSharedValue(false);
+
+    const attachmentCarouselPagerContext = useContext(AttachmentCarouselPagerContext);
+    const {
+        isUsedInCarousel,
+        isSingleCarouselItem,
+        isPagerScrolling,
+        page,
+        activePage,
+        onTap,
+        onScaleChanged: onScaleChangedContext,
+        pagerRef,
+    } = useMemo(() => {
```

Comment:
I'm not really certain what the purpose of this memo is at all. Can you help me understand it, please?

Reply:
@tgolen yes, sure. I think the `useMemo` was used mainly because of two reasons:
- to avoid computation of `findIndex`/iteration over array during re-render;
- to avoid re-rendering of underlying views in case if `attachmentCarouselPagerContext` is `null` (otherwise `onTap` and `onScaleChanged` will be re-created on every re-render and it will cause unnecessary re-render of `MultiGestureCanvas`)

So the usage of `useMemo` should improve performance 🚀

Reply:
OK, thanks!